compiler/optimizer/VPHandlers.cpp

```diff
@@ -4241,13 +4241,65 @@ TR::Node *constrainNew(OMR::ValuePropagation *vp, TR::Node *node)
          vp->addGlobalConstraint(node, classConstraint->asClass()->getClassType());
       else
          vp->addGlobalConstraint(node, classConstraint);
+
+      // If it's an abstract class or an interface, the allocation cannot be removed
+      // because InstantiationError could be thrown
+      TR_OpaqueClassBlock *clazz = classConstraint->getClassType() ? classConstraint->getClassType()->getClass() : NULL;
+      if (clazz && TR::Compiler->cls.isConcreteClass(vp->comp(), clazz))
```

Comment:
Thanks Henry! Good catch! I'll add `!TR::Compiler->cls.isValueTypeClass(clazz)` in this condition check. Just double checking, since I know you're updating the code distinguishing between primitive value types and ordinary value types in the JIT, will ClassEnv::isValueTypeClass() still return true for both value types? If not, what is the new API? I think for either primitive value types or ordinary value types, `new` should result in an `InstantiationError`.

Reply:
Addressed in f2616d4